auth-api/src/auth_api/resources/user.py

```diff
@@ -32,12 +33,68 @@
 from auth_api.utils.roles import Role, Status
 from auth_api.utils.util import cors_preflight
 
-
 API = Namespace('users', description='Endpoints for user profile management')
 TRACER = Tracer.get_instance()
 _JWT = JWTWrapper.get_instance()
 
 
+@cors_preflight('POST,OPTIONS')
+@API.route('/<invitation_token>', methods=['POST', 'OPTIONS'])
+class AnonymousUser(Resource):
+    """Resource for managing anonymous users."""
+
+    @staticmethod
+    @TRACER.trace()
+    @cors.crossdomain(origin='*')
+    def post(invitation_token):
+        """Post a new user using the request body who has a proper invitation."""
+        try:
+            invitation = InvitationService.validate_token(invitation_token).as_dict()
+
+            valid_format, errors = schema_utils.validate(request.get_json(), 'anonymous_user')
+            if not valid_format:
+                return {'message': schema_utils.serialize(errors)}, http_status.HTTP_400_BAD_REQUEST
+
+            membership_details = {
+                'email': invitation['recipientEmail'],
+                'membershipType': invitation['membership'][0]['membershipType']
+            }
+            membership_details.update(request.get_json())
+            user = UserService.create_user_and_add_membership([membership_details],
+                                                              invitation['membership'][0]['org']['id'], skip_auth=True)
+            InvitationService.accept_invitation(invitation['id'], None, None, False)
+            response, status = user, http_status.HTTP_201_CREATED
+
+        except BusinessException as exception:
+            response, status = {'code': exception.code, 'message': exception.message}, exception.status_code
+        return response, status
+
+
+@cors_preflight('POST,OPTIONS')
+@API.route('/bulk', methods=['POST', 'OPTIONS'])
```

Comment:
Normally the last path in the endpoint is a resource. So to stick with standards, can we have the endpoint as /bulk/<resources> ?